packages/typegpu/src/name.ts

```diff
@@ -0,0 +1,53 @@
+import { $getNameForward } from './shared/symbols.ts';
+
+interface MetaData {
+  name: string | undefined;
+}
+
+declare global {
+  var __TYPEGPU_META__: WeakMap<object, MetaData>;
+}
+
+function isObject(value: unknown): value is object {
+  return !!value && (typeof value === 'object' || typeof value === 'function');
+}
+
+function isForwarded(value: object): value is { [$getNameForward]: object } {
+  if (!($getNameForward in value)) {
+    return false;
+  }
+  return isObject(value[$getNameForward]);
+}
```

Comment:
Since we're checking if the value is an object again at the beginning of `getName` we can omit it.
```suggestion
function isForwarded(value: object): value is { [$getNameForward]: unknown } {
  return !!value?.[$getNameForward];
}
```